tuf/client/updater.py

```diff
@@ -2971,8 +2971,24 @@ def _visit_child_role(self, child_role, target_filepath):
     child_role_name = child_role['name']
     child_role_paths = child_role.get('paths')
     child_role_path_hash_prefixes = child_role.get('path_hash_prefixes')
+    child_role_delegation_hash_prefix_len = child_role.get(
+        'delegation_hash_prefix_len')
 
-    if child_role_path_hash_prefixes is not None:
+    if child_role_delegation_hash_prefix_len is not None:
+      target_filepath_hash = self._get_target_hash(target_filepath)
+      # Find the bin associated with this hash and return the bin name
+      prefix_bits = child_role_delegation_hash_prefix_len
+      number_of_bins = 2 ** prefix_bits
+      prefix_length = len("{:x}".format(number_of_bins - 1))
+      bin_size = 16 ** prefix_length // number_of_bins
+      prefix = int(target_filepath_hash[:prefix_length], 16)
+
+      bin_num = (prefix - (prefix % bin_size)) / bin_size
+
+      bin_name = child_role_name + "{num:0{len}x}".format(num=int(bin_num), len=prefix_length)
```

Comment:
```suggestion
      bin_name = "{role_name}{num:0{len}x}".format(
          role_name=child_role_name, num=int(bin_num), len=prefix_length)
```
We should try and avoid mixing string concatenation and string formatting